src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -1936,7 +1936,34 @@ private Type RetrieveClrTypeForConstant(IEdmTypeReference edmTypeReference, ref
                 Contract.Assert(strValue != null);
 
                 constantType = Nullable.GetUnderlyingType(constantType) ?? constantType;
-                value = Enum.Parse(constantType, strValue);
+
+                IEdmEnumType enumType = edmTypeReference.AsEnum().EnumDefinition();
+                ClrEnumMemberAnnotation memberMapAnnotation = Model.GetClrEnumMemberAnnotation(enumType);
+                if (memberMapAnnotation != null)
+                {
+                    IEdmEnumMember enumMember = enumType.Members.FirstOrDefault(m => m.Name == strValue);
+                    if (enumMember == null)
+                    {
+                        enumMember = enumType.Members.FirstOrDefault(m => m.Value.ToString() == strValue);
+                    }
+
+                    if (enumMember != null)
+                    {
+                        Enum clrMember = memberMapAnnotation.GetClrEnumMember(enumMember);
+                        if (clrMember != null)
```

Comment:
>if (clrMember != null) [](start = 24, length = 22)

what if clrMember is null here?  it looks like value is never set?

Reply:
It's set in the Model builder when we add the annotation.  If it's null, It's better to throw exception. Let me add

---
In reply to: [485247896](https://github.com/OData/WebApi/pull/2267#discussion_r485247896) [](ancestors = 485247896)